src/Data/Vector/Mutable/Linear.hs

```diff
@@ -132,6 +174,64 @@ read (Vec size' arr) ix
         & \(arr', val) -> (Vec size' arr', val)
   | otherwise = arr `lseq` error "Read index not in range."
 
+-- | Same as 'read', but does not do bounds-checking. The behaviour is undefined
+-- when passed an invalid index.
+unsafeRead :: HasCallStack => Vector a #-> Int -> (Vector a, Unrestricted a)
```

Comment:
For consistency, could you change `read` and `write` to use the same elegant design as the arrays that call `unsafeRead` and `unsafeWrite` with `assertIndexInRange ix arr` that panics if the index is not in range?